src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Integer.Signed.Default.cs

```diff
@@ -2,47 +2,166 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
 namespace System.Buffers.Text
 {
     /// <summary>
     /// Methods to format common data types as Utf8 strings.
     /// </summary>
     public static partial class Utf8Formatter
     {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool TryFormatInt64Default(long value, Span<byte> buffer, out int bytesWritten)
         {
             if ((ulong)value < 10)
             {
-                if (buffer.Length == 0) goto FalseExit;
-                buffer[0] = (byte)('0' + value);
-                bytesWritten = 1;
+                return TryFormatUInt32SingleDigit((uint)value, buffer, out bytesWritten);
+            }
+#if BIT32
+            if (value <= int.MaxValue && value >= int.MinValue)
+            {
+                return TryFormatInt32MultipleDigits((int)value, buffer, out bytesWritten);
+            }
+            else
+            {
+                if (value <= Utf8Constants.Quintillion && value >= -Utf8Constants.Quintillion)
+                {
+                    return TryFormatInt64LessThanQuintillion(value, buffer, out bytesWritten);
+                }
+                else
+                {
+                    return TryFormatInt64MoreThanQuintillion(value, buffer, out bytesWritten);
+                }
+            }
+#else
+            return TryFormatInt64MultipleDigits(value, buffer, out bytesWritten);
+#endif // BIT32
+        }
+
+        // TODO: Use this instead of TryFormatInt64Default to format numbers less than int.MaxValue for BIT32
```

Comment:
I think you should use this to format int32, even on 64-bit platforms.

Reply:
Fair enough. That is what I was planning to do. Will tackle this in a future PR (or file an issue if I can't get to it).

Reply:
I will update the comment though.